pywbemtools/pywbemcli/_connection_repository.py

```diff
@@ -460,79 +544,84 @@ def add(self, svr_definition):
-    def _open_file(filename, file_mode='w'):
+    def _open_file(filename, file_mode):
         """
-        A static convenience function that performs the open of the connection
-        definitions file correctly for different versions of Python.
+        Open the specified file in text mode with UTF-8 encoding.
 
-        This covers the issue where the file should be opened in text mode but
-        that is done differently in Python 2 and Python 3.
+        This method encapsulates a difference between Python 2 and Python 3
+        when opening files in text mode with UTF-8 encoding.
 
         The returned file-like object must be closed by the caller.
 
         Parameters:
 
           filename (:term:`string`):
-            Name of the file where the recorder output will be written
+            Path name of the file to be opened.
 
           file_mode (:term:`string`):
-            Optional file mode.  The default is 'w' which overwrites any
-            existing file.  if 'a' is used, the data is appended to any
-            existing file.
+            File mode, for example:
+            - 'r' opens the file for reading.
+            - 'w' opens the file for writing and truncates an existing file.
+            Must be a text mode, i.e. 'b' is not permitted in the file mode.
 
         Returns:
 
           File-like object.
+
+        Raises:
+          OSError
+          IOError (on Python 2, and possibly for lower level errors)
         """
+        assert 'b' not in file_mode
         if six.PY2:
-            # Open with codecs to define text mode
             return codecs.open(filename, mode=file_mode, encoding='utf-8')
-
         return open(filename, file_mode, encoding='utf8')
 
-    def _write_connections_file(self):  # pylint: disable=no-self-use
+    def _write_connections_file(self):
         """
-        Write the connections file if one has been loaded.
-        If the dictionary is empty, it deletes the file.
+        Write the connection repository state to the connections file.
 
-        If there is an existing file it is moved to the same name with the
-        suffix .bak and a new current file written.
+        If there is an existing connections file, it is backed up with a suffix
+        of '.bak' before writing the connections file.
 
         Raises:
-            ConnectionsFileWriteError: If the file cannot be written.
+          ConnectionsFileWriteError
         """
         conn_dict = {}
         if self._pywbemcli_servers:
             conn_dict = {name: value.to_dict() for name, value in
                          self._pywbemcli_servers.items()}
 
             # build dictionary for YAML output
-            yaml_dict = {ConnectionRepository.connections_group_name: conn_dict,
-                         ConnectionRepository.default_connection_grp_name:
-                             self.default_connection_name}
+            yaml_dict = {
+                self.connections_group_name: conn_dict,
```

Comment:
I would prefer to use the classname here because these are class level variables

Reply:
Class level variables are officially also available on the object (i.e. self). It makes it shorter, but I'm open either way.

Reply:
DONE After discussing with Karl, I have changed it back to using the class name.